### PR TITLE
Feat: Upgrade to tailwind 4

### PR DESCRIPTION
**Pull request by ItsMeBrianD**
### Checklist

- [ ] For UI or styling changes, I have added a screenshot or gif showing before & after
- [ ] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [ ] I have added to the docs where applicable
- [ ] I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable

### CODE REVIEW

**Comment by archiewood** [comment]
there are some useful looking chromatic test failures here

File: packages/ui/core-components/src/app.css [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  If we ever want to remove these styles, we need to add an explicit border
+  color utility to any element that depends on these defaults.
+*/
+@layer base {
+	*,
+	::after,
+	::before,
+	::backdrop,
+	::file-selector-button {
+		border-color: var(--color-gray-200, currentColor);
```
Comment (archiewood) [inline packages/ui/core-components/src/app.css:19]: is this a valid color?

**Comment by archiewood** [comment]
I think this is currently choking on the docs deploy because we use and inline style

`[@tailwindcss/vite:generate:build] Cannot apply unknown utility class: border-gray-200`


CleanShot 2025-01-27 at 17 59 50@2x

File: packages/ui/tailwind/src/config/createVarsForColors/createVarsForColors.spec.js [hunk 2]
```diff
@@ -24,17 +10,35 @@ describe('createVarsForColors', () => {
...
-    --someColorToken: someColorValue-dark
+html.theme-dark {
+	--color-someColorToken: someColorValue-dark;
+	--someColorToken: someColorValue-dark;
 }
 `.trim()
 
-		const actual = await runPlugin(themes);
-		expect(actual).toEqual(expected);
+		// const actual = await runPlugin(themes);
```
Comment (zachstence) [inline packages/ui/tailwind/src/config/createVarsForColors/createVarsForColors.spec.js:41]:
```suggestion
```

**Review by zachstence — CHANGES_REQUESTED** [review]

File: packages/lib/universal-sql/src/build-parquet.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ import { emptyDbFs, initDB, query } from './client-duckdb/node.js';
 import chunk from 'lodash.chunk';
 import { columnsToScore } from './calculateScore.js';
 import chalk from 'chalk';
-import { log } from '@evidence-dev/sdk/logger';
+// import { log } from '@evidence-dev/sdk/logger';
```
[the remark this answers is not in the document]
Comment (ItsMeBrianD) [reply packages/lib/universal-sql/src/build-parquet.js:23]: Yes, this removes the dependency on the sdk for universal sql (seemed to be causing issues, or at least a ton of noise)

File: e2e/hmr/tests/tests.spec.js [hunk 4]
```diff
@@ -27,7 +27,14 @@ test.describe('pages', () => {
...
 	});
 
-	test('creating should add to the sidebar and allow navigation', async ({ page }) => {
+	/*
+		The following tests are skipped because of a vite behavior where a page is ineligible for
+		HMR if the dev server hasn't yet transformed it. (e.g. the user hasn't opened it)
+
+		While this isn't a behavior we want, it is enough of an edge case to not spend a lot of 
+		time fixing up for now.
+	*/
```
[the remark this answers is not in the document]
Comment (zachstence) [reply e2e/hmr/tests/tests.spec.js]: https://github.com/evidence-dev/evidence/issues/3093

**Review by zachstence — APPROVED** [review]
